camping.py

```
@@ -266,76 +281,15 @@ def main(parks, json_output=False):
-    parser.add_argument("--debug", "-d", action="store_true", help="Debug log level")
-    parser.add_argument(
-        "--start-date", required=True, help="Start date [YYYY-MM-DD]", type=valid_date
-    )
-    parser.add_argument(
-        "--end-date",
-        required=True,
-        help="End date [YYYY-MM-DD]. You expect to leave this day, not stay the night.",
-        type=valid_date,
-    )
-    parser.add_argument(
-        "--nights",
-        help="Number of consecutive nights (default is all nights in the given range).",
-        type=positive_int,
-    )
-    parser.add_argument(
-        "--campsite-type",
-        help=(
-            "If you want to filter by a type of campsite. For example "
-            '"STANDARD NONELECTRIC" or TODO'
-        ),
-    )
-    parser.add_argument(
-        "--json-output",
-        action="store_true",
-        help=(
-            "This make the script output JSON instead of human readable "
-            "output. Note, this is incompatible with the twitter notifier. "
-            "This output includes more precise information, such as the exact "
-            "avaiable dates and which sites are available."
-        ),
-    )
-    parks_group = parser.add_mutually_exclusive_group(required=True)
-    parks_group.add_argument(
-        "--parks", dest="parks", metavar="park", nargs="+", help="Park ID(s)", type=int
-    )
-    parks_group.add_argument(
-        "--stdin",
-        "-",
-        action="store_true",
-        help="Read list of park ID(s) from stdin instead",
-    )
-
+    parser = CampingArgParser()
     args = parser.parse_args()
 
     if args.debug:
         LOG.setLevel(logging.DEBUG)
 
-    parks = args.parks or [p.strip() for p in sys.stdin]
-
     try:
-        code = 0 if main(parks, json_output=args.json_output) else 1
+        code = 0 if main(args.parks, json_output=args.json_output) else 1
         sys.exit(code)
     except Exception:
         print("Something went wrong")
```

While we're here just remove this try / except. I'd rather see the exception than have it swallowed like this.

---

Done.